Amplify/Categories/Storage/StorageCategory+ClientBehavior.swift

```diff
@@ -90,6 +90,12 @@ extension StorageCategory: StorageCategoryBehavior {
         try await plugin.getURL(key: key, options: options)
     }
 
+    @discardableResult
+    public func downloadData(key: String,
+                      options: StorageDownloadDataOperation.Request.Options?) async throws -> StorageDownloadDataTask {
+        try await plugin.downloadData(key: key, options: options)
```

Comment:
you may be confusing the protocol from this file, where the protocol doesn't have default parameters. StorageCategory` methods can and does have the `nil` default parameter for optional parameters, for example, the previous downloadData API on line 20 has the default parameter but this new async API does not. (https://github.com/aws-amplify/amplify-ios/issues/2223)

Reply:
nit: indentation

Reply:
yes, missed the indentation on this one. Once the async additions are in I'll be removing the listener functions. I can then make these other requested changes.